### PR TITLE
Feature/issue 9 create company

### PR DESCRIPTION
**Pull request by zxander-D**
issue 9

### CODE REVIEW

File: src/main/java/com/hundred/percent/capstone/Invoicify/company/controller/CompanyController.java [hunk 1]
```diff
@@ -1,21 +1,38 @@
...
+    @Autowired
+    CompanyService companyService;
+
+    /*@GetMapping
     public String getCompanies(){
         return "[]";
+    }*/
+
+    @GetMapping
+    public List<CompanyEntity> getCompanies(){
```
Comment (ranjanr2) [inline src/main/java/com/hundred/percent/capstone/Invoicify/company/controller/CompanyController.java]: This should be a company DTO. Company Entity is being used here. Please check

File: src/main/java/com/hundred/percent/capstone/Invoicify/company/service/CompanyService.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+public class CompanyService {
+
+    @Autowired
+    CompanyRepository companyRepository;
+
+    public void createCompany(CompanyDTO companyDTO) {
+        companyRepository.save(new CompanyEntity(companyDTO.getName(), companyDTO.getAddress(), companyDTO.getContact_name(), companyDTO.getContact_title(), companyDTO.getContact_phone_number()));
+    }
+
+    public List<CompanyEntity> getAllCompanies(){
```
Comment (ranjanr2) [inline src/main/java/com/hundred/percent/capstone/Invoicify/company/service/CompanyService.java]: This should be a company DTO returning out of the service layer

File: src/test/java/com/hundred/percent/capstone/Invoicify/company/CompanyIT.java [hunk 3]
```diff
@@ -46,4 +47,37 @@ public void postCompanyTest() throws Exception {
...
+
+        mockMvc.perform(post("/companies/addCompany")
+                .content(objectMapper.writeValueAsString(input2))
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isCreated())
+                .andDo(print());
+
+        mockMvc.perform(get("/companies"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("length()").value(2))
```
Comment (ranjanr2) [inline src/test/java/com/hundred/percent/capstone/Invoicify/company/CompanyIT.java:71]: A company is supposed to have an company invoice number. please check

**Review by ranjanr2 — CHANGES_REQUESTED** [review]
Please see the comments. Also please if you can get latest from Main.

**Review by sunita1501 — COMMENTED** [review]
Review comments added by Rohit. Please modify the code.

File: src/main/java/com/hundred/percent/capstone/Invoicify/company/controller/CompanyController.java (hunk 1 above)
Comment (404-bug-not-found) [reply src/main/java/com/hundred/percent/capstone/Invoicify/company/controller/CompanyController.java]: We specifically wants to return entity, so response can have ID field.

**Review by ranjanr2 — APPROVED** [review]

**Review by sunita1501 — APPROVED** [review]

**Pull request merged**